frontend/packages/client/src/components/Community/SubscribeCommunityButton.js

```diff
@@ -0,0 +1,75 @@
+import { useModalContext } from 'contexts/NotificationModal';
+import { useNotificationServiceContext } from 'contexts/NotificationService';
+import { useWebContext } from 'contexts/Web3';
+import { Svg } from '@cast/shared-components';
+import { ErrorModal, NotificationsModal, WalletConnect } from 'components';
+import classnames from 'classnames';
+
+export default function SubscribeCommunityButton({
+  communityId,
+  className,
+  size = 'small',
+}) {
+  const { openModal, closeModal } = useModalContext();
+  const { notificationSettings } = useNotificationServiceContext();
+  const isSubscribed = notificationSettings?.communitySubscription.some(
+    (c) => c === communityId
```

Comment:
here we need check for the higher level user preference `isSubscribedFromCommunityUpdates` too.
```suggestion
    const isSubscribed = notificationSettings.isSubscribedFromCommunityUpdates && <community level Subscription is true>
```
If `isSubscribedFromCommunityUpdates` is false, during subscription it need invoke resubscribe method too to enable Email: https://github.com/DapperCollectives/CAST/blob/8cbc14696f9369c690b31d918f917ad85f035d94/frontend/packages/client/src/contexts/NotificationService.js#L115

Reply:
if im understanding correctly this should be the logic we want:

https://github.com/DapperCollectives/CAST/pull/650/commits/e0e75e01547c99707f216cda0239a10d1d7c1ec4